src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java

```diff
@@ -0,0 +1,99 @@
+    String newPassword = getNewPassword();
+
+    public void userLogsIn() {
+        logInPageObject.getLogInLink().click();
+        logInPageObject.getEmailInputField().sendKeys(propertyChangePassword.getPropertyValue("passUserName"));
+        logInPageObject.getPasswordInputField().sendKeys(propertyChangePassword.getPropertyValue("currentPassword"));
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void navigateToChangePasswordPage() {
+        wait.until(ExpectedConditions.visibilityOf(changePasswordObjects.getUserEmail()));
+        changePasswordObjects.getUserEmail().click();
+        changePasswordObjects.getChangePassword().click();
+    }
+
+    public void typeInOldPassword() {
+        changePasswordObjects.getOldPasswordField().sendKeys(propertyChangePassword.getPropertyValue("currentPassword"));
+    }
+
+    public void typeInNewPassword() throws IOException {
+        changePasswordObjects.getNewPasswordField().sendKeys(newPassword);
+        changePasswordObjects.getConfirmPasswordField().sendKeys(newPassword);
+        writePass(newPassword);
+    }
+
+    public void clickChangePasswordButton() {
+        changePasswordObjects.getChangePasswordButton().click();
+    }
+
+    public void checkConfirmationMessageIsDisplayed() {
+        wait.until(ExpectedConditions.visibilityOf(changePasswordObjects.getChangePasswordConfirmation()));
+        assertThat("Change password message displayed", changePasswordObjects.getChangePasswordConfirmation().getText(), is(LogInValidationMessages.CHANGE_PASSWORD_CONFIRMATION.getValidationMessage()));
+    }
+
+    public void logInWithOldPassword() {
+        logInPageObject.getLogOutLink().click();
+        logInPageObject.getLogInLink().click();
+        logInPageObject.getEmailInputField().sendKeys(propertyChangePassword.getPropertyValue("passUserName"));
+        logInPageObject.getPasswordInputField().sendKeys(propertyChangePassword.getPropertyValue("oldPassword"));
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void checkWrongCredentialsMessage() {
+        assertThat("User cannot log in with old password", logInPageObject.getWrongCredentialsMessage().getText(), is(LogInValidationMessages.WRONG_CREDENTIALS.getValidationMessage()));
+    }
+
+    public void LogInWithNewPassword() {
+        logInPageObject.getLogInLink().click();
+        logInPageObject.getEmailInputField().sendKeys(propertyChangePassword.getPropertyValue("passUserName"));
+        logInPageObject.getPasswordInputField().sendKeys(newPassword);
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void checkThatUserIsLoggedIn() {
+        assertThat("LogIn was successful", checkAccountNameIsDisplayed(), is(propertyChangePassword.getPropertyValue("passUserName")));
+    }
+
+    public void writePass(String value) throws IOException {
+        Path filePath = Paths.get("src/main/resources/configuration/ChangePasswordCredentials.properties");
```

Comment:
Can be simplified by adding in PropertyReader another method to set property (setProperty()), but it's up to you